src/crypto/tests/CHIPCryptoPALTest.cpp

```diff
@@ -49,9 +49,45 @@
 #include <stdlib.h>
 #include <string.h>
 
+#define HSM_ECC_KEYID 0x11223344
+
 using namespace chip;
 using namespace chip::Crypto;
 
+class Test_P256Keypair :
```

Comment:
consider a single ifdef:

```cpp

#ifdef ENABLE_HSM_EC_KEY

class Test_P256Keypair : public P256KeypairHSM {
   public:
      Test_P256Keypair() { SetKeyId(HSM_ECC_KEY_ID); }
      Test_P256Keypair(int keyId) { SetKeyId(keyId); }
};

using TestSpake2p_P256_SHA256_HKDF_HMAC = Spake2pHSM_P256_SHA256_HKDF_HMAC;

#else 

using Test_P256Keypair = P256Keypair;
using TestSpake2p_P256_SHA256_HKDF_HMAC = Spake2p_P256_SHA256_HKDF_HMAC;

#endif // using TestSpake2p_P256_SHA256_HKDF_HMAC
```

seems easier to understand.

Reply:
changed accordingly